clustered/common/src/main/java/org/ehcache/clustered/common/messages/EhcacheEntityResponse.java

```diff
@@ -82,10 +113,14 @@ public static Failure failure(Exception cause) {
 
     private final Exception cause;
 
-    private Failure(Exception cause) {
+    Failure(Exception cause) {
       this.cause = cause;
     }
 
+    Failure(ByteBuffer failure) {
+      this.cause = (Exception)Util.unmarshall(failure.array());
```

Comment:
This looks dodgy to me: calling `ByteBuffer.array()` only works with heap byte buffers which you don't know if you have such one here.